compiler/src/test/java/com/squareup/anvil/compiler/codegen/ContributesSubcomponentHandlerGeneratorTest.kt

```diff
@@ -1060,9 +1083,128 @@ class ContributesSubcomponentHandlerGeneratorTest {
     }
   }
 
-  private val Class<*>.anvilComponent: Class<*>
-    get() = classLoader
-      .loadClass("$COMPONENT_PACKAGE_PREFIX.${generatedClassesString()}$ANVIL_SUBCOMPONENT_SUFFIX")
+  @Test fun `subcomponent can be contributed and bindings replaced in a 2nd compilation`() {
+    // This test simulates a compilation in the main source set and test/androidTest source set,
+    // where contributed subcomponents are generated a second time for the test components. This
+    // test ensures that there are no duplicate generated classes.
+    val firstResult = compile(
+      """
+        package com.squareup.test
+
+        import com.squareup.anvil.annotations.ContributesSubcomponent
+        import com.squareup.anvil.annotations.ContributesTo
+        import com.squareup.anvil.annotations.MergeComponent
+        import dagger.Module
+        import dagger.Provides
+
+        @ContributesSubcomponent(Any::class, parentScope = Unit::class)
+        interface SubcomponentInterface {
+          fun integer(): Int
+        }
+
+        @ContributesTo(Any::class)
+        @Module
+        object DaggerModule1 {
+          @Provides fun provideIntegerFive(): Int = 5
+        }
+
+        @MergeComponent(Unit::class)
+        interface ComponentInterface1
+      """,
+      enableDaggerAnnotationProcessor = true
+    ) {
+      val daggerComponent = componentInterface1.daggerComponent.declaredMethods
+        .single { it.name == "create" }
+        .invoke(null)
+
+      val subcomponent = componentInterface1.methods
+        .single()
+        .invoke(daggerComponent)
+
+      val int = subcomponent::class.java.declaredMethods
+        .single { it.name == "integer" }
+        .use { it.invoke(subcomponent) as Int }
+
+      assertThat(int).isEqualTo(5)
+    }
+
+    compile(
+      """
+        package com.squareup.test
+
+        import com.squareup.anvil.annotations.ContributesSubcomponent
+        import com.squareup.anvil.annotations.ContributesTo
+        import com.squareup.anvil.annotations.MergeComponent
+        import dagger.Module
+        import dagger.Provides
+
+        @ContributesTo(Any::class, replaces = [DaggerModule1::class])
+        @Module
+        object DaggerModule2 {
+          @Provides fun provideIntegerSix(): Int = 6
```

Comment:
Notice that this Dagger modules replaces the other Dagger modules and the new generated subcomponent picks up the replaced module. That's quite common in tests.